crates/client/db/src/l1_db.rs

```diff
@@ -128,4 +129,32 @@ impl MadaraBackend {
         self.db.put_cf_opt(&nonce_column, bincode::serialize(&nonce)?, /* empty value */ [], &writeopts)?;
         Ok(())
     }
+
+    pub fn get_l1_handler_tx_hashes(&self, l1_tx_hash: TxHash) -> Result<Vec<Felt>, DbError> {
+        let l1_l2_mappings_column = self.db.get_column(Column::L1MessagingHandlerTxHashes);
+        let mut l1_handler_tx_hashes = vec![];
+        for kv_bytes in self.db.prefix_iterator_cf(&l1_l2_mappings_column, l1_tx_hash.as_slice()) {
+            let l1_handler_tx_hash = Felt::from_bytes_be_slice(&kv_bytes?.1);
+            l1_handler_tx_hashes.push(l1_handler_tx_hash);
+        }
+        Ok(l1_handler_tx_hashes)
+    }
+
+    /// Store mapping from L1 transaction to L1 handler transaction (on the L2). A unique order
+    /// value is required to ensure the handler transactions are retreived in the correct order.
+    pub fn add_l1_handler_tx_hash_mapping(
+        &self,
+        l1_tx_hash: TxHash,
+        l1_handler_tx_hash: Felt,
+        order: u64,
+    ) -> Result<(), DbError> {
+        let l1_l2_mappings_column = self.db.get_column(Column::L1MessagingHandlerTxHashes);
+        let mut key = [0u8; 40];
+        key[..32].copy_from_slice(l1_tx_hash.as_slice());
+        key[32..].copy_from_slice(&order.to_be_bytes()); // BE is important for the lexographic sorting
+        let mut writeopts = WriteOptions::default();
+        writeopts.disable_wal(true);
```

Comment:
Shouldn't WAL be enabled so we don't risk loosing this on a node restart?

Reply:
Storing the mapping is guarded behind `messaging_update_last_synced_l1_block_with_event`, which also has WAL disabled. The sync will restart from the last stored `LastSyncedEventBlock`. So I think that should also be enabled? @cchudant